src/parser.rs

```diff
@@ -47,17 +51,78 @@ where
     }
 }
 
-pub fn parse_meta_command(input: &str) -> Result<Option<MetaCommand>, ParserError> {
+pub fn parse_meta_command(input: &str) -> MetaCommand {
     if input.starts_with('.') {
+        if input.starts_with(".createdb") {
+            match parse_createdb(input) {
+                Ok(createdb_meta_command) => return createdb_meta_command,
+                Err(error) => return MetaCommand::MetacommandWithWrongArgs(MetaCommandError::ParseError(error.to_string())),
+            }
+        } else if input.starts_with(".dropdb") {
+            match parse_dropdb(input) {
+                Ok(dropdb_meta_command) => return dropdb_meta_command,
+                Err(error) => return MetaCommand::MetacommandWithWrongArgs(MetaCommandError::ParseError(error.to_string())),
+            }
+        }
+
         match input {
-            ".exit" | ".quit" => Ok(Some(MetaCommand::Exit)),
-            _ => Err(ParserError::UnknownMetaCommand(input)),
+            ".exit" | ".quit" => MetaCommand::Exit,
+            _ => MetaCommand::Unknown(input.to_string()),
         }
     } else {
-        Ok(None)
+        MetaCommand::Void
     }
 }
 
+pub fn parse_createdb(input: &str) -> Result<MetaCommand, ParserError> {
+    let current_dir = PathBuf::from("./");
+
+    let mut input_iterator = input.splitn(3, ' ');
+    input_iterator.next(); // skip ".createdb"
+
+    let db_path_str =
+        pathify(
+            input_iterator.next().ok_or(ParserError::DatabaseNameMissing)?,
+            &current_dir
+        );
+
+    let db_path = PathBuf::from(&db_path_str);
+    let db_file_name = db_path
+        .file_name().ok_or(ParserError::CouldNotParseDbFilename(input))?
+        .to_str().ok_or(ParserError::CouldNotParseDbFilename(input))?;
+
+    let db_dir_path = db_path.parent()
+        .ok_or(ParserError::CouldNotParseDbFilename(input))?;
+
+    let tables_dir_path_str = match input_iterator.next() {
+        Some(string) => pathify(string, &current_dir),
+        None => format!("{}/{}_tables/",
+                        db_dir_path.display(),
+                        db_file_name),
```

Comment:
- probably `_tables` suffix should a const
- instead of concat, probably one can use `push(format!("{}_tables", db_file_name))`